### ISSUE TITLE
Infinite loop and enormous memory consumption when building release

### ISSUE THREAD

**Issue by khuey** [rust-lang/rust#50323]
Check out https://github.com/khuey/cpp_demangle/tree/experimental.

`cargo build` works fine.
`cargo build --release` seems to go into an iloop and burn through memory (I gave it 10 minutes of CPU and 16GB of memory before I killed it).

The previous commit works fine, so something in https://github.com/khuey/cpp_demangle/commit/df43ff095c44325e0753b55cf47854de5deb32f6 is triggering it.

I see this on both stable and latest nightly.
```
rustc -V --verbose
rustc 1.25.0 (84203cac6 2018-03-25)
binary: rustc
commit-hash: 84203cac67e65ca8640b8392348411098c856985
commit-date: 2018-03-25
host: x86_64-unknown-linux-gnu
release: 1.25.0
LLVM version: 6.0
```
```
rustc -V --verbose
rustc 1.27.0-nightly (66363b288 2018-04-28)
binary: rustc
commit-hash: 66363b288141ed2af4b95fb4205a56dce6248363
commit-date: 2018-04-28
host: x86_64-unknown-linux-gnu
release: 1.27.0-nightly
LLVM version: 6.0
```

**Comment by ExpHP** [comment]
Possibly related to https://github.com/rust-lang/rust/issues/49402

**Comment by fitzgen** [comment]
Looks like this is all in LLVM:

![screen shot 2018-04-30 at 9 18 06 am](https://user-images.githubusercontent.com/74571/39438003-7d8f90fc-4c57-11e8-97ae-94a2a4458edd.png)

**Comment by ExpHP** [comment]
Here's a self-contained test case that just barely hits the sweet spot where it starts to grow out of hand:

```rust
fn main() {
    use ::std::collections::HashSet;
    let set: HashSet<_> = {
        (0..86).into_iter()
            .chain(87..89)
            .chain(92..93)
            .chain(94..105)
//            .chain(107..108)
            .chain(116..118)
            .chain(122..124)
            .chain(128..131)
            .chain(133..134)
//            .chain(141..143)
            .chain(151..152)
            .chain(153..154)
//            .chain(158..168)
//            .chain(169..170)
//            .chain(179..180)
//            .chain(181..182)
//            .chain(185..186)
            .chain(188..189)
            .chain(201..202)
//            .chain(203..204)
            .collect()
    };
    assert!(set.contains(&153));
}
```

I do observe that "linking" is the last thing printed... but not quite in the same sense as @fitzgen: :stuck_out_tongue:

```
    ...
    time: 51.579; rss: 174MB    codegen passes [whee1-faa1bfdce812aee96e5ffecf666ddd72.rs]
  time: 61.759; rss: 154MB      LLVM passes
  time: 0.000; rss: 151MB       serialize work products
    time: 0.250; rss: 150MB     running linker
  time: 0.250; rss: 150MB       linking
    Finished release [optimized] target(s) in 62.30 secs
$
```

<details>
<summary>cargo +nightly rustc --release -- -Ztime-passes (62.3s)</summary>

```
$ cargo +nightly rustc --release -- -Ztime-passes
   Compiling whee v0.1.0 (file:///home/lampam/asd/clone/cpp_demangle/whee)
  time: 0.000; rss: 46MB        parsing
  time: 0.000; rss: 46MB        recursion limit
  time: 0.000; rss: 46MB        crate injection
  time: 0.000; rss: 46MB        plugin loading
  time: 0.000; rss: 46MB        plugin registration
    time: 0.020; rss: 71MB      expand crate
    time: 0.000; rss: 71MB      check unused macros
  time: 0.020; rss: 71MB        expansion
  time: 0.000; rss: 71MB        maybe building test harness
  time: 0.000; rss: 71MB        maybe creating a macro crate
  time: 0.000; rss: 71MB        creating allocators
  time: 0.000; rss: 71MB        AST validation
  time: 0.002; rss: 72MB        name resolution
  time: 0.000; rss: 72MB        complete gated feature checking
  time: 0.001; rss: 73MB        lowering ast -> hir
  time: 0.000; rss: 73MB        early lint checks
  time: 0.000; rss: 73MB        indexing hir
  time: 0.000; rss: 73MB        load query result cache
  time: 0.000; rss: 73MB        looking for entry point
  time: 0.000; rss: 73MB        looking for plugin registrar
  time: 0.000; rss: 73MB        loop checking
  time: 0.000; rss: 76MB        attribute checking
  time: 0.000; rss: 76MB        stability checking
  time: 0.000; rss: 76MB        type collecting
  time: 0.000; rss: 76MB        outlives testing
  time: 0.000; rss: 76MB        impl wf inference
  time: 0.000; rss: 76MB        coherence checking
  time: 0.000; rss: 76MB        variance testing
  time: 0.000; rss: 80MB        wf checking
  time: 0.000; rss: 80MB        item-types checking
  time: 0.048; rss: 120MB       item-bodies checking
  time: 0.003; rss: 121MB       rvalue promotion
  time: 0.000; rss: 121MB       privacy checking
  time: 0.000; rss: 121MB       intrinsic checking
  time: 0.000; rss: 122MB       match checking
  time: 0.000; rss: 122MB       liveness checking
  time: 0.005; rss: 123MB       borrow checking
  time: 0.000; rss: 123MB       MIR borrow checking
  time: 0.000; rss: 123MB       dumping chalk-like clauses
  time: 0.000; rss: 123MB       MIR effect checking
  time: 0.000; rss: 123MB       death checking
  time: 0.000; rss: 123MB       unused lib feature checking
  time: 0.000; rss: 123MB       lint checking
  time: 0.000; rss: 123MB       resolving dependency formats
    time: 0.000; rss: 123MB     write metadata
    time: 0.056; rss: 133MB     translation item collection
    time: 0.001; rss: 133MB     codegen unit partitioning
    time: 0.000; rss: 134MB     write allocator module
    time: 0.015; rss: 142MB     llvm function passes [whee1]
    time: 0.003; rss: 146MB     llvm function passes [whee13]
    time: 0.017; rss: 146MB     llvm function passes [whee0]
    time: 0.004; rss: 148MB     llvm function passes [whee4]
    time: 0.022; rss: 149MB     llvm module passes [whee4]
    time: 0.007; rss: 149MB     llvm function passes [whee3]
    time: 0.021; rss: 150MB     llvm module passes [whee3]
    time: 0.092; rss: 150MB     llvm module passes [whee13]
    time: 0.007; rss: 150MB     llvm function passes [whee5]
    time: 0.002; rss: 150MB     llvm function passes [whee2]
    time: 0.011; rss: 150MB     llvm module passes [whee2]
    time: 0.002; rss: 151MB     llvm function passes [whee7]
    time: 0.006; rss: 151MB     llvm module passes [whee7]
    time: 0.002; rss: 152MB     llvm function passes [whee9]
    time: 0.011; rss: 152MB     llvm module passes [whee9]
    time: 0.002; rss: 152MB     llvm function passes [whee11]
    time: 0.056; rss: 152MB     llvm module passes [whee5]
    time: 0.008; rss: 152MB     llvm module passes [whee11]
    time: 0.001; rss: 152MB     llvm function passes [whee10]
    time: 0.002; rss: 153MB     llvm function passes [whee6]
    time: 0.004; rss: 153MB     llvm module passes [whee10]
    time: 0.007; rss: 153MB     llvm module passes [whee6]
    time: 0.002; rss: 154MB     llvm function passes [whee8]
    time: 0.003; rss: 154MB     llvm module passes [whee8]
    time: 0.001; rss: 154MB     llvm function passes [whee12]
    time: 0.090; rss: 154MB     translate to LLVM IR
    time: 0.000; rss: 154MB     assert dep graph
    time: 0.000; rss: 154MB     serialize dep graph
  time: 0.288; rss: 154MB       translation
    time: 0.002; rss: 153MB     llvm module passes [whee12]
    time: 0.001; rss: 151MB     llvm function passes [whee14]
    time: 0.001; rss: 152MB     llvm module passes [whee14]
    time: 0.001; rss: 152MB     llvm function passes [whee15]
    time: 0.002; rss: 152MB     llvm module passes [whee15]
    time: 0.190; rss: 152MB     llvm module passes [whee0]
    time: 6.645; rss: 183MB     llvm module passes [whee1]
    time: 0.021; rss: 159MB     LTO passes
    time: 0.013; rss: 161MB     codegen passes [whee9-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.036; rss: 160MB     LTO passes
    time: 0.003; rss: 161MB     LTO passes
    time: 0.004; rss: 162MB     codegen passes [whee12-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.015; rss: 164MB     codegen passes [whee5-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.011; rss: 164MB     LTO passes
    time: 0.007; rss: 165MB     LTO passes
    time: 0.012; rss: 167MB     codegen passes [whee2-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.007; rss: 167MB     codegen passes [whee4-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.073; rss: 166MB     LTO passes
    time: 0.008; rss: 170MB     LTO passes
    time: 0.020; rss: 168MB     LTO passes
    time: 0.008; rss: 168MB     codegen passes [whee7-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.010; rss: 167MB     codegen passes [whee13-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.006; rss: 167MB     LTO passes
    time: 0.004; rss: 167MB     codegen passes [whee3-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.004; rss: 168MB     LTO passes
    time: 0.004; rss: 171MB     codegen passes [whee10-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.005; rss: 172MB     LTO passes
    time: 0.003; rss: 173MB     LTO passes
    time: 0.006; rss: 173MB     codegen passes [whee6-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.004; rss: 173MB     codegen passes [whee15-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.002; rss: 173MB     LTO passes
    time: 0.003; rss: 174MB     codegen passes [whee8-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.060; rss: 173MB     codegen passes [whee0-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.004; rss: 170MB     LTO passes
    time: 0.001; rss: 170MB     LTO passes
    time: 0.002; rss: 170MB     codegen passes [whee11-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 0.001; rss: 170MB     codegen passes [whee14-faa1bfdce812aee96e5ffecf666ddd72.rs]
    time: 3.249; rss: 180MB     LTO passes
    time: 51.579; rss: 174MB    codegen passes [whee1-faa1bfdce812aee96e5ffecf666ddd72.rs]
  time: 61.759; rss: 154MB      LLVM passes
  time: 0.000; rss: 151MB       serialize work products
    time: 0.250; rss: 150MB     running linker
  time: 0.250; rss: 150MB       linking
    Finished release [optimized] target(s) in 62.30 secs
```

</details>

<details>
<summary>cargo +nightly rustc --release -- -Ztime-passes -Ccodegen-units=1 (63.2s)</summary>

```
$ cargo +nightly rustc --release -- -Ztime-passes -Ccodegen-units=1
   Compiling whee v0.1.0 (file:///home/lampam/asd/clone/cpp_demangle/whee)
  time: 0.000; rss: 46MB        parsing
  time: 0.000; rss: 46MB        recursion limit
  time: 0.000; rss: 46MB        crate injection
  time: 0.000; rss: 46MB        plugin loading
  time: 0.000; rss: 46MB        plugin registration
    time: 0.017; rss: 71MB      expand crate
    time: 0.000; rss: 71MB      check unused macros
  time: 0.017; rss: 71MB        expansion
  time: 0.000; rss: 71MB        maybe building test harness
  time: 0.000; rss: 71MB        maybe creating a macro crate
  time: 0.000; rss: 71MB        creating allocators
  time: 0.000; rss: 71MB        AST validation
  time: 0.001; rss: 72MB        name resolution
  time: 0.000; rss: 72MB        complete gated feature checking
  time: 0.000; rss: 74MB        lowering ast -> hir
  time: 0.000; rss: 74MB        early lint checks
  time: 0.000; rss: 74MB        indexing hir
  time: 0.000; rss: 74MB        load query result cache
  time: 0.000; rss: 74MB        looking for entry point
  time: 0.000; rss: 74MB        looking for plugin registrar
  time: 0.000; rss: 74MB        loop checking
  time: 0.000; rss: 77MB        attribute checking
  time: 0.000; rss: 77MB        stability checking
  time: 0.000; rss: 77MB        type collecting
  time: 0.000; rss: 77MB        outlives testing
  time: 0.000; rss: 77MB        impl wf inference
  time: 0.000; rss: 77MB        coherence checking
  time: 0.000; rss: 77MB        variance testing
  time: 0.000; rss: 77MB        wf checking
  time: 0.000; rss: 77MB        item-types checking
  time: 0.047; rss: 120MB       item-bodies checking
  time: 0.003; rss: 121MB       rvalue promotion
  time: 0.000; rss: 121MB       privacy checking
  time: 0.000; rss: 121MB       intrinsic checking
  time: 0.000; rss: 122MB       match checking
  time: 0.000; rss: 122MB       liveness checking
  time: 0.005; rss: 124MB       borrow checking
  time: 0.000; rss: 124MB       MIR borrow checking
  time: 0.000; rss: 124MB       dumping chalk-like clauses
  time: 0.000; rss: 124MB       MIR effect checking
  time: 0.000; rss: 124MB       death checking
  time: 0.000; rss: 124MB       unused lib feature checking
  time: 0.000; rss: 124MB       lint checking
  time: 0.000; rss: 124MB       resolving dependency formats
    time: 0.000; rss: 124MB     write metadata
    time: 0.057; rss: 134MB     translation item collection
    time: 0.001; rss: 134MB     codegen unit partitioning
    time: 0.000; rss: 134MB     write allocator module
    time: 0.060; rss: 143MB     translate to LLVM IR
    time: 0.000; rss: 143MB     assert dep graph
    time: 0.000; rss: 143MB     serialize dep graph
  time: 0.124; rss: 143MB       translation
    time: 0.034; rss: 142MB     llvm function passes [whee0]
    time: 6.867; rss: 172MB     llvm module passes [whee0]
    time: 55.740; rss: 165MB    codegen passes [whee0]
  time: 62.665; rss: 144MB      LLVM passes
  time: 0.000; rss: 144MB       serialize work products
    time: 0.245; rss: 144MB     running linker
  time: 0.245; rss: 144MB       linking
    Finished release [optimized] target(s) in 63.23 secs
```

</details>

---

```
$ rustc +nightly --version
rustc 1.27.0-nightly (79252ff4e 2018-04-29)
```

**Comment by JustAPerson** [comment]
Seems to have been resolved. Works fine with a recent nightly to compile @ExpHP's example (in addition, with everything uncommented). Appears to have been fixed a while ago as rustc 2019-09-23 works fine too.

<details>
<summary>cargo +nightly rustc --release -- -Ztime-passes</summary>

```
warning: unused manifest key: profiles
   Compiling issue50323 v0.1.0 (/tmp/issue50323)
time: 0.000; rss: 50MB	parse_crate
time: 0.000; rss: 50MB	attributes_injection
time: 0.000; rss: 50MB	recursion_limit
time: 0.000; rss: 50MB	plugin_loading
time: 0.000; rss: 50MB	plugin_registration
time: 0.000; rss: 50MB	pre_AST_expansion_lint_checks
time: 0.000; rss: 54MB	crate_injection
time: 0.013; rss: 68MB	expand_crate
time: 0.000; rss: 68MB	check_unused_macros
time: 0.013; rss: 68MB	macro_expand_crate
time: 0.000; rss: 68MB	maybe_building_test_harness
time: 0.000; rss: 68MB	AST_validation
time: 0.000; rss: 68MB	maybe_create_a_macro_crate
time: 0.000; rss: 71MB	complete_gated_feature_checking
time: 0.014; rss: 71MB	configure_and_expand
time: 0.000; rss: 71MB	prepare_outputs
time: 0.000; rss: 74MB	hir_lowering
time: 0.000; rss: 74MB	early_lint_checks
time: 0.000; rss: 77MB	validate_HIR_map
time: 0.000; rss: 77MB	setup_global_ctxt
time: 0.000; rss: 77MB	dep_graph_tcx_init
time: 0.001; rss: 77MB	create_global_ctxt
time: 0.000; rss: 77MB	looking_for_entry_point
time: 0.000; rss: 77MB	looking_for_plugin_registrar
time: 0.000; rss: 77MB	looking_for_derive_registrar
time: 0.000; rss: 80MB	misc_checking_1
time: 0.000; rss: 80MB	type_collecting
time: 0.000; rss: 80MB	impl_wf_inference
time: 0.000; rss: 80MB	unsafety_checking
time: 0.000; rss: 80MB	orphan_checking
time: 0.000; rss: 80MB	coherence_checking
time: 0.000; rss: 84MB	wf_checking
time: 0.000; rss: 84MB	item_types_checking
time: 0.028; rss: 99MB	item_bodies_checking
time: 0.028; rss: 99MB	type_check_crate
time: 0.002; rss: 103MB	match_checking
time: 0.000; rss: 103MB	liveness_and_intrinsic_checking
time: 0.003; rss: 103MB	misc_checking_2
time: 0.018; rss: 108MB	MIR_borrow_checking
time: 0.000; rss: 108MB	dumping_chalk_like_clauses
time: 0.000; rss: 108MB	MIR_effect_checking
time: 0.000; rss: 108MB	layout_testing
time: 0.000; rss: 108MB	death_checking
time: 0.000; rss: 108MB	unused_lib_feature_checking
time: 0.000; rss: 108MB	crate_lints
time: 0.000; rss: 108MB	module_lints
time: 0.000; rss: 108MB	lint_checking
time: 0.000; rss: 108MB	privacy_checking_modules
time: 0.001; rss: 108MB	misc_checking_3
time: 0.000; rss: 108MB	monomorphization_collector_root_collections
time: 0.056; rss: 123MB	monomorphization_collector_graph_walk
time: 0.004; rss: 123MB	partition_and_assert_distinct_symbols
time: 0.000; rss: 124MB	write_allocator_module
time: 0.000; rss: 124MB	find_cgu_reuse
time: 0.008; rss: 132MB	llvm function passes [issue50323.aepfjml6-cgu.0]
time: 0.005; rss: 134MB	llvm function passes [issue50323.aepfjml6-cgu.2]
time: 0.002; rss: 138MB	llvm function passes [issue50323.aepfjml6-cgu.5]
time: 0.002; rss: 138MB	llvm function passes [issue50323.aepfjml6-cgu.12]
time: 0.007; rss: 138MB	llvm function passes [issue50323.aepfjml6-cgu.1]
time: 0.008; rss: 140MB	llvm module passes [issue50323.aepfjml6-cgu.5]
time: 0.003; rss: 141MB	llvm function passes [issue50323.aepfjml6-cgu.3]
time: 0.001; rss: 141MB	llvm function passes [issue50323.aepfjml6-cgu.7]
time: 0.006; rss: 141MB	llvm module passes [issue50323.aepfjml6-cgu.7]
time: 0.005; rss: 142MB	llvm function passes [issue50323.aepfjml6-cgu.4]
time: 0.001; rss: 142MB	llvm function passes [issue50323.aepfjml6-cgu.11]
time: 0.001; rss: 142MB	llvm module passes [issue50323.aepfjml6-cgu.11]
time: 0.001; rss: 143MB	llvm function passes [issue50323.aepfjml6-cgu.6]
time: 0.001; rss: 143MB	llvm function passes [issue50323.aepfjml6-cgu.8]
time: 0.003; rss: 144MB	llvm module passes [issue50323.aepfjml6-cgu.8]
time: 0.014; rss: 144MB	llvm module passes [issue50323.aepfjml6-cgu.4]
time: 0.021; rss: 144MB	llvm module passes [issue50323.aepfjml6-cgu.3]
time: 0.001; rss: 144MB	llvm function passes [issue50323.aepfjml6-cgu.9]
time: 0.000; rss: 144MB	llvm module passes [issue50323.aepfjml6-cgu.9]
time: 0.087; rss: 144MB	codegen_to_LLVM_IR
time: 0.000; rss: 144MB	assert_dep_graph
time: 0.000; rss: 144MB	serialize_dep_graph
time: 0.164; rss: 144MB	codegen_crate
time: 0.008; rss: 126MB	llvm module passes [issue50323.aepfjml6-cgu.6]
time: 0.000; rss: 115MB	llvm function passes [issue50323.aepfjml6-cgu.13]
time: 0.003; rss: 115MB	free_global_ctxt
time: 0.001; rss: 115MB	llvm module passes [issue50323.aepfjml6-cgu.13]
time: 0.000; rss: 115MB	llvm function passes [issue50323.aepfjml6-cgu.14]
time: 0.000; rss: 115MB	llvm function passes [issue50323.aepfjml6-cgu.10]
time: 0.000; rss: 115MB	llvm module passes [issue50323.aepfjml6-cgu.14]
time: 0.000; rss: 115MB	llvm function passes [issue50323.aepfjml6-cgu.15]
time: 0.001; rss: 116MB	llvm module passes [issue50323.aepfjml6-cgu.15]
time: 0.001; rss: 116MB	llvm module passes [issue50323.aepfjml6-cgu.10]
time: 0.049; rss: 116MB	llvm module passes [issue50323.aepfjml6-cgu.12]
time: 0.134; rss: 118MB	llvm module passes [issue50323.aepfjml6-cgu.2]
time: 0.143; rss: 118MB	llvm module passes [issue50323.aepfjml6-cgu.0]
time: 0.213; rss: 118MB	llvm module passes [issue50323.aepfjml6-cgu.1]
time: 0.003; rss: 119MB	LTO_passes
time: 0.007; rss: 120MB	codegen passes [issue50323.aepfjml6-cgu.6]
time: 0.013; rss: 122MB	LTO_passes
time: 0.002; rss: 122MB	LTO_passes
time: 0.016; rss: 123MB	LTO_passes
time: 0.016; rss: 123MB	LTO_passes
time: 0.004; rss: 123MB	codegen passes [issue50323.aepfjml6-cgu.8]
time: 0.007; rss: 123MB	codegen passes [issue50323.aepfjml6-cgu.5]
time: 0.001; rss: 123MB	LTO_passes
time: 0.001; rss: 123MB	LTO_passes
time: 0.002; rss: 123MB	codegen passes [issue50323.aepfjml6-cgu.10]
time: 0.009; rss: 123MB	codegen passes [issue50323.aepfjml6-cgu.12]
time: 0.002; rss: 124MB	codegen passes [issue50323.aepfjml6-cgu.7]
time: 0.001; rss: 124MB	LTO_passes
time: 0.001; rss: 124MB	LTO_passes
time: 0.002; rss: 124MB	codegen passes [issue50323.aepfjml6-cgu.9]
time: 0.002; rss: 124MB	codegen passes [issue50323.aepfjml6-cgu.13]
time: 0.002; rss: 124MB	LTO_passes
time: 0.001; rss: 124MB	LTO_passes
time: 0.002; rss: 124MB	codegen passes [issue50323.aepfjml6-cgu.15]
time: 0.001; rss: 124MB	LTO_passes
time: 0.002; rss: 125MB	codegen passes [issue50323.aepfjml6-cgu.11]
time: 0.001; rss: 125MB	codegen passes [issue50323.aepfjml6-cgu.14]
time: 0.021; rss: 125MB	codegen passes [issue50323.aepfjml6-cgu.4]
time: 0.045; rss: 126MB	LTO_passes
time: 0.029; rss: 127MB	codegen passes [issue50323.aepfjml6-cgu.1]
time: 0.086; rss: 128MB	LTO_passes
time: 0.116; rss: 128MB	LTO_passes
time: 0.133; rss: 128MB	LTO_passes
time: 0.048; rss: 128MB	codegen passes [issue50323.aepfjml6-cgu.2]
time: 0.019; rss: 128MB	codegen passes [issue50323.aepfjml6-cgu.3]
time: 0.041; rss: 129MB	codegen passes [issue50323.aepfjml6-cgu.0]
time: 0.456; rss: 129MB	LLVM_passes
time: 0.000; rss: 130MB	join_worker_thread
time: 0.366; rss: 130MB	finish_ongoing_codegen
time: 0.000; rss: 130MB	serialize_work_products
time: 0.000; rss: 130MB	link_binary_check_files_are_writeable
time: 0.112; rss: 130MB	run_linker
time: 0.000; rss: 130MB	link_binary_remove_temps
time: 0.113; rss: 130MB	link_binary
time: 0.113; rss: 130MB	link_crate
time: 0.000; rss: 130MB	llvm_dump_timing_file
time: 0.479; rss: 130MB	link
time: 0.716; rss: 130MB		total
    Finished release [optimized] target(s) in 0.83s
```
</details>

**Comment by khuey** [comment]
This went away at some point, yeah.
